client/src/SignupNonProfit.tsx

```diff
@@ -59,129 +101,184 @@ function SignupNonProfit() {
+                                        label="Role Title"
+                                        placeholder="Role Title"
+                                        value={formData.role_or_title}
+                                        onChange={handleFieldChange}
+                                    />
+                                </Grid>
+                                <Grid item md={6} xs={12}>
+                                    <TextField
+                                        id="email"
+                                        label="Email"
+                                        placeholder="Email"
+                                        value={formData.email}
+                                        onChange={handleFieldChange}
+                                    />                            
+                                </Grid>
+                                <Grid item md={6} xs={12}>
+                                    <TextField
+                                        id="password"
+                                        label="Password"
+                                        placeholder="Password"
+                                        value={formData.password}
+                                        onChange={handleFieldChange}
+                                    />                            
+                                </Grid>
+                                    <FormControlLabel
+                                        style={{ textAlign: 'left', display: 'block' }}
+                                        control={
+                                            <Checkbox
+                                                color="primary"
+                                                checked={formData.accept_terms}
+                                                onChange={handleFieldChange}
+                                                name="accept_terms"
+                                                inputProps={{ 'aria-label': 'accept_terms_checkbox' }}
+                                            />
+                                        }
+                                        label={
+                                            <label>
+                                                Accept the{' '}
+                                                <StyledLink to="/terms_and_condtions" target="_blank">
+                                                    Terms and Condtions
+                                                </StyledLink>
+                                            </label>
+                                        }
+                                    />
+                                    <Button
+                                        className={classes.button}
+                                        fullWidth
+                                        type="submit"
+                                        disabled={!formData.accept_terms}
```

Comment:
check that all inputs are filled out
if it get's too long just make it a variable above the return
`const step1Complete = formData.a && formData.b ....`
`const step2Complete = step1Complete && formData.y && formData.z ....`

Reply:
And then I put the firstPageComplete validator in the button on the first page (just fyi)